### ISSUE TITLE
feature request: BoosterPipeline

### ISSUE THREAD

**Issue by koaning** [koaning/scikit-lego#39]
The idea is to have a pipeline where you might have more than one model in sequention. Model 2 would try to improve on the residuals of Model 1 and so forth.

**Comment by MaxHalford** [comment]
Not sure this is what you're looking for but I wrote a generic gradient library called [starboost](https://github.com/MaxHalford/starboost) from which you might want to inspire yourself!

**Comment by koaning** [comment]
The idea here is to implement a pipeline like;

```
from sklego.transformers import RandomAdder

from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForest 

...

mod = BoosterPipeline([
    ("scale", StandardScaler()),
    ("random_noise", RandomAdder()),
    ("global_model", LogisticRegression(solver='lbfgs')), 
    ("residual_model", RandomForest())
])

...
```

It's more of a UI thing.

**Comment by MaxHalford** [comment]
So it is still like some form of stacking? Are you going to reuse the features that the global used, or are you going to base yourself on the predictions?

**Comment by koaning** [comment]
This is a good question. I might argue that: 

1. this could be a setting
2. the base behaviour should probably be that you don't use the predictions from a previous model, rather that you the same features and that you had before (depending on other transformations)

**Comment by koaning** [comment]
I am wondering if this feature might break up the workflow too much. Instead it might also work if we just have a `Booster`-model that accepts many different models.

**Comment by MBrouns** [comment]
I tried to build something like this quite a while back and remember it took a lot of changes in the normal pipeline behaviour. A `Booster` meta-model might be simpler
